Classes/EventStore/Normalizer/ValueObjectNormalizer.php

```diff
@@ -44,6 +47,11 @@ public function supportsDenormalization($data, $className, $format = null): bool
 
     private function resolveConstructorMethod(string $dataType, string $className): ReflectionMethod
     {
+        $cacheIdentifier = md5($dataType . $className);
```

Comment:
To prevent the (frankly very unlikely) case of overlaps we should add a separator between the cache id parts:
```suggestion
        $cacheIdentifier = md5($dataType . '|' . $className);
```